src/Libraries/Microsoft.Extensions.Http.Resilience/Internal/RequestMessageSnapshot.cs

```diff
@@ -64,24 +86,76 @@ bool IResettable.TryReset()
         _method = null;
         _version = null;
         _requestUri = null;
+        if (_content is StreamContent)
+        {
+            // a snapshot's StreamContent is always a unique copy (deep clone)
+            // therefore, it is safe to dispose when snapshot is no longer needed
+            _content.Dispose();
+        }
+
         _content = null;
 
         return true;
     }
 
     void IDisposable.Dispose() => _snapshots.Return(this);
 
-    private void Initialize(HttpRequestMessage request)
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Resilience", "EA0014:The async method doesn't support cancellation", Justification = "Past the point of no cancellation.")]
+    private static async Task<(HttpContent? content, HttpContent? clonedContent)> CloneContentAsync(HttpContent? content)
     {
-        if (request.Content is StreamContent)
+        HttpContent? clonedContent = null;
+        if (content != null)
         {
-            Throw.InvalidOperationException($"{nameof(StreamContent)} content cannot by cloned.");
+            HttpContent originalContent = content;
+            Stream originalRequestBody = await content.ReadAsStreamAsync().ConfigureAwait(false);
+            MemoryStream clonedRequestBody = new MemoryStream();
+            await originalRequestBody.CopyToAsync(clonedRequestBody).ConfigureAwait(false);
+            clonedRequestBody.Position = 0;
+            if (originalRequestBody.CanSeek)
+            {
+                originalRequestBody.Position = 0;
+            }
+            else
+            {
+                originalRequestBody = new MemoryStream();
+                await clonedRequestBody.CopyToAsync(originalRequestBody).ConfigureAwait(false);
+                originalRequestBody.Position = 0;
+                clonedRequestBody.Position = 0;
+            }
+
+            clonedContent = new StreamContent(clonedRequestBody);
+            content = new StreamContent(originalRequestBody);
+            foreach (KeyValuePair<string, IEnumerable<string>> header in originalContent.Headers)
+            {
+                _ = clonedContent.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                _ = content.Headers.TryAddWithoutValidation(header.Key, header.Value);
+            }
         }
 
+        return (content, clonedContent);
+    }
+
+    private bool IsReset()
+    {
+        return _method == null;
+    }
```

Comment:
Should be this be a property?

Reply:
It's private, so no; however, I did refactor this to `IsInitialized()`. It eliminates ambiguity since technically an instance can be uninitialized without ever having been reset.